sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileClient.cs

```diff
@@ -2715,6 +2761,46 @@ Response<Blobs.Models.BlobDownloadInfo> response
                 conditions: overwrite ? null : new DataLakeRequestConditions { IfNoneMatch = new ETag(Constants.Wildcard) },
                 cancellationToken: cancellationToken);
 
+        /// <summary>
+        /// The <see cref="UploadAsync(Stream, DataLakeFileUploadOptions, CancellationToken)"/>
+        /// operation creates and uploads content to a file.  If the file already exists, its content will be overwriten.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update" />.
+        /// </summary>
+        /// <param name="content">
+        /// A <see cref="Stream"/> containing the content to upload.
+        /// </param>
+        /// <param name="options">
+        /// Optional parameters.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{BlobContentInfo}"/> describing the
+        /// state of the updated block blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual Task<Response<PathInfo>> UploadAsync(
+            Stream content,
+            DataLakeFileUploadOptions options,
+            CancellationToken cancellationToken = default) =>
+            StagedUploadAsync(
+                content,
+                options?.HttpHeaders,
+                options?.Metadata,
+                options?.Permissions,
+                options?.Umask,
+                options?.Conditions,
+                options?.ProgressHandler,
+                options.TransferOptions,
```

Comment:
".?"

Reply:
Should we add test cases where options are not provided?

Reply:
No, cause options is required.  I'll remove the .?, it's a habit.

Reply:
Do we want to let pass `null` ?